presto-hive/src/main/java/io/prestosql/plugin/hive/AcidInfo.java

```diff
@@ -272,7 +273,15 @@ public Builder addDeleteDelta(Path deleteDeltaPath, long minWriteId, long maxWri
                     deleteDeltaPath.getParent().toString(),
                     partitionLocation);
 
-            deleteDeltaInfos.add(new DeleteDeltaInfo(minWriteId, maxWriteId, statementId));
+             // Handling different formats of Delete delta file:
+             // 1. delete_delta_minWriteId_maxWriteID_statementId: default case
+             // 2. delete_delta_minWriteId_maxWriteID: after minor compaction
+            if (deleteDeltaPath.getName().split("_").length == 5) {
```

Comment:
Why are we parsing the `deleteDeltaPath` if `statementId` is given as a parameter here?

Reply:
We fetch this ID from AcidUtils which has the following logic:
```
public int getStatementId() {
            return statementId == -1 ? 0 : statementId;
 }
```
So, we get 0 even if it is -1, that's why we need to figure it out from delete delta file name.

Reply:
I would prefer to create a shim over `AcidUtils.getAcidState` and do the postprocessing immediatelly after calling original method. So logic is not scattered around. WDYT?